examples/thermomechanics/thermal_deformation_heat_transfer.cpp

```diff
@@ -0,0 +1,162 @@
+    std::array<double, 3> high_corner = inputs["high_corner"];
+    std::array<int, 3> num_cells = inputs["num_cells"];
+    int m = std::floor( delta /
+                        ( ( high_corner[0] - low_corner[0] ) / num_cells[0] ) );
+    int halo_width = m + 1; // Just to be safe.
+
+    // ====================================================
+    //                Force model type
+    // ====================================================
+    using model_type = CabanaPD::PMB;
+    using thermal_type = CabanaPD::DynamicTemperature;
+
+    // ====================================================
+    //                 Particle generation
+    // ====================================================
+    // Does not set displacements, velocities, etc.
+    auto particles =
+        std::make_shared<CabanaPD::Particles<memory_space, model_type,
+                                             typename thermal_type::base_type>>(
+            exec_space(), low_corner, high_corner, num_cells, halo_width );
+
+    // ====================================================
+    //            Custom particle initialization
+    // ====================================================
+    auto rho = particles->sliceDensity();
+    auto temp = particles->sliceTemperature();
+    auto init_functor = KOKKOS_LAMBDA( const int pid )
+    {
+        // Density
+        rho( pid ) = rho0;
+        // Temperature
+        temp( pid ) = temp0;
+    };
+    particles->updateParticles( exec_space{}, init_functor );
+
+    // ====================================================
+    //                    Force model
+    // ====================================================
+    auto force_model = CabanaPD::createForceModel(
+        model_type{}, CabanaPD::Elastic{}, *particles, delta, K, kappa, cp,
+        alpha, temp0 );
+
+    // ====================================================
+    //                   Create solver
+    // ====================================================
+    auto cabana_pd = CabanaPD::createSolverElastic<memory_space>(
+        inputs, particles, force_model );
+
+    // ====================================================
+    //                   Boundary condition
+    // ====================================================
+    // Temperature profile imposed on top, bottom, left, and right surfaces
+    double dy = particles->dx[1];
+    using plane_type = CabanaPD::RegionBoundary<CabanaPD::RectangularPrism>;
+
+    // Top surface: y-direction
+    plane_type plane1( low_corner[0], high_corner[0], high_corner[1] - dy,
+                       high_corner[1] + dy, low_corner[2], high_corner[2] );
+
+    // Bottom surface: y-direction
+    plane_type plane2( low_corner[0], high_corner[0], low_corner[1] - dy,
+                       low_corner[1] + dy, low_corner[2], high_corner[2] );
+
+    std::vector<plane_type> planes = { plane1, plane2 };
+    temp = particles->sliceTemperature();
```

Comment:
Why do we need this here. Note we have:
Line 83: auto temp = particles->sliceTemperature();
Line 89: temp( pid ) = temp0;